### PR TITLE
[refactor] implement dotenv to hide api-keys from codebase

### LINKED ISSUE #273: The Api Keys and Database Credentials  are clearly visible
### 🐛 Describe the bug

I navigate through the project today, Pretty easy to setup. 
But when overlooking the project realised that the api keys and database credentials are easily accessible which is kind of risky and potentially the project could be in risk because of it. 

I would be interested if can have .env file which has all the api credentials and if someone really want we can have a drive link of the real api credentials which can be given access when someone requests it.
Also the other way someone can create own firebase and appwrite project and do all the testing in their own environment.

### PR DESCRIPTION
**Pull request by opxdelwin**
## Description

This PR will add a `.env` file to securely store these credentials. The code now retrieves them from the environment variable, keeping them hidden within the development environment.

Here's what this means:
 - Improved security: Credentials are no longer directly exposed in the code.
 - Separate testing setup: Developers can easily create their own Firebase and Appwrite projects for local testing without needing real credentials.

Fixes #273 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (does not change functionality, e.g. code style improvements, linting)
- [x] Documentation update

## How Has This Been Tested?

I haven't added unit tests as these are code refactors, moreover it doesn't look like tests have been initialized beforehand.
I'll be happy to add if required.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxx
- [ ] Tag the PR with the appropriate labels

### CODE REVIEW

File: docs/GETTING_STARTED.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    ```bash
+    flutter pub get
+    ```
+
+4. **Configure environment variables:**
+
+    - Duplicate `env/.env.sample` and rename as `.env`
+    - Create a Firebase project [here](https://console.firebase.google.com/)
+    - Set up firebase app as mentioned [here]()
```
Comment (chandansgowda) [inline docs/GETTING_STARTED.md:35]: Add link here

**Review by chandansgowda — CHANGES_REQUESTED** [review]
Can't we just tell them to add the firebase_options.dart file instead of loading it from env file ?

File: docs/GETTING_STARTED.md (hunk 1 above)
Comment (opxdelwin) [reply docs/GETTING_STARTED.md:35]: Yes will do!

**Comment by opxdelwin** [comment]
As of now, API keys of the production server are exposed, which might have been unintended. This could lead to exploitation of resources and incur unexpected costs. That is why we prefer .env while initialisation of any project.

Secondly, But then, later when we're to run automation to publish releases straight to the respective platform, or run checks on PR and commits, removing this file from the codebase would raise issues.

File: docs/GETTING_STARTED.md (hunk 1 above)
Comment (opxdelwin) [reply docs/GETTING_STARTED.md:35]: I've updated, along with non null checks for optional parameters on `firebase_options.dart`.

**Comment by Aarush-Acharya** [comment]
Hey @opxdelwin Thanks a lot for your contribution  🚀✨,

having a .env is always considered as on of the best practices for any repository. But currenlty we planning to have a .bat script that creates an appwrite env similar to our main env on the contibutors local machine using appwrite self hosted (docker). Thus we wont have to expose our env variables anymore rather more control would be in the contributors hands.

We really appreciate your work towards Resonate. There are a lot of issues and things to be done in the app.  Lets move forward and build Resonate a beautiful app. 

Thank you

**Pull request closed without merging**